pkg/controller/machineset/controller.go

```diff
@@ -251,6 +251,10 @@ func (r *ReconcileMachineSet) reconcile(ctx context.Context, machineSet *cluster
 		return reconcile.Result{}, errors.Wrap(err, "failed to update machine set status")
 	}
 
+	if syncErr != nil {
+		return reconcile.Result{}, errors.Wrapf(syncErr, "failed to sync machines")
```

Comment:
Any reason we shouldn't check the error just after the call to `syncReplicas`?

Reply:
@vincepri so that machineset status status could get updated and along with error message in the events, user could have latest stats.